File: Assets/Bundles/Shared/Resources/TS/UserInput/Drivers/MobileJoystickDriver.lua
-- Compiled with unity-ts v2.1.0-75
local Signal = require("Shared/TS/Util/Signal").Signal
local MobileJoystickDriver
do
	MobileJoystickDriver = setmetatable({}, {
		__tostring = function()
			return "MobileJoystickDriver"
		end,
	})
	MobileJoystickDriver.__index = MobileJoystickDriver
	function MobileJoystickDriver.new(...)
		local self = setmetatable({}, MobileJoystickDriver)
		return self:constructor(...) or self
	end
	function MobileJoystickDriver:constructor()
		self.Changed = Signal.new()
		UserInputService.InputProxy:OnMobileJoystickEvent(function(position, phase)
			self.Changed:Fire(position, phase)
		end)
	end
	function MobileJoystickDriver:SetVisible(visible)
		UserInputService.InputProxy:SetMobileJoystickVisible(visible)
	end
	function MobileJoystickDriver:IsVisible()
		return UserInputService.InputProxy:IsMobileJoystickVisible()
	end
	function MobileJoystickDriver:instance()
		if self.inst == nil then
			self.inst = MobileJoystickDriver.new()
		end
		return self.inst
	end
end
return {
	MobileJoystickDriver = MobileJoystickDriver,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/@Easy/Core/Core~/src/Shared/UserInput/Drivers/MobileJoystickDriver.ts
import { Signal } from "@Easy/Core/Shared/Util/Signal";

export class MobileJoystickDriver {
	private static inst: MobileJoystickDriver;

	public readonly changed = new Signal<[position: Vector3, phase: MobileJoystickPhase]>();

	private constructor() {
		// UserInputService.InputProxy.OnMobileJoystickEvent((position, phase) => {
		// 	this.Changed.Fire(position, phase);
		// });
	}

	public SetVisible(visible: boolean) {
		// InputBridge.Instance.SetMobileJoystickVisible(visible);
	}

	public IsVisible() {
		// return InputBridge.Instance.IsMobileJoystickVisible();
	}

	/** **NOTE:** Internal only. Use `Touchscreen` class instead. */
	public static Instance() {
		return (this.inst ??= new MobileJoystickDriver());
	}
}
